src/main/scala/com/fulcrumgenomics/bam/ErrorRateByReadPosition.scala

```diff
@@ -25,266 +25,234 @@
-    ref match {
-      case Some(r) => new ErrorRateByReadPositionWithReferenceCounter(ref=r, includeInsertions=includeInsertions, minBaseQuality=minBaseQuality)
-      case None    => new ErrorRateByReadPositionWithMdCounter(includeInsertions=includeInsertions, minBaseQuality=minBaseQuality)
-    }
-  }
-}
+          val readBase = SequenceUtil.upperCase(rec.getReadBase)
+          val readNum = readNumber(rec.getRecord)
+          val cycle = if (rec.getRecord.getReadNegativeStrandFlag) rec.getRecord.getReadLength - rec.getOffset else rec.getOffset + 1
+          val counter = counters(readNum).getOrElseUpdate(cycle, new ObsCounter(readNum, cycle))
 
-/** Base class for computing the error rate by position.
-  *
-  * Sub-classes should implement the [updateErrors] method to extract the read bases and qualities, and reference bases.
-  * They should then call [incrementIfValid] or [increment] to update the counters for each read offset.
-  *
-  * @param minBaseQuality the minimum base quality to allow.
-  */
-private abstract class ErrorRateByReadPositionCounter(includeInsertions: Boolean, minBaseQuality: Int) {
-  private val errorsByReadOffset = List.tabulate(3)(x => new NumericCounter[Int]())
-  private val totalByReadOffset  = List.tabulate(3)(x => new NumericCounter[Int]())
+          val (ref, read) = {
+            if (refBase == 'A' || refBase == 'C') (refBase.toChar, readBase.toChar)
+            else (SequenceUtil.complement(refBase).toChar, SequenceUtil.complement(readBase).toChar)
+          }
 
-  /** Gets the list of metrics sorted by read number and then position. */
-  def metrics: Seq[ErrorRateByReadPositionMetric] = {
-    if (totalByReadOffset.forall(_.isEmpty)) return Seq.empty
-    // create the metrics
-    errorsByReadOffset.zip(totalByReadOffset).zipWithIndex.filter {
-      case ((errCounter, totalCounter), idx) => totalCounter.nonEmpty
-    }.flatMap { case ((errCounter, totalCounter), idx) =>
-      val maxOffset = totalCounter.map(_._1).max
-      Range.inclusive(0, maxOffset).map { offset =>
-        val error_bases = errCounter.countOf(offset)
-        val error_total = totalCounter.countOf(offset)
-        val error_rate  = if (0 == error_total) 0 else error_bases / error_total.toDouble
-        new ErrorRateByReadPositionMetric(
-          read_number = idx,
-          position    = offset + 1,
-          error_rate  = error_rate,
-          bases_total = error_total
-        )
+          (ref, read) match {
+            case ('A', 'A') => counter.a_ref_obs  += 1
+            case ('A', 'C') => counter.a_to_c_obs += 1
+            case ('A', 'G') => counter.a_to_g_obs += 1
+            case ('A', 'T') => counter.a_to_t_obs += 1
+            case ('C', 'A') => counter.c_to_a_obs += 1
+            case ('C', 'C') => counter.c_ref_obs  += 1
+            case ('C', 'G') => counter.a_to_g_obs += 1
+            case ('C', 'T') => counter.c_to_t_obs += 1
+          }
```

Comment:
How hard would it be to add support for insertions as errors?  I think the locus iterator has support for gathering indels, but I don't know how much effort it would be to add.